meinberlin/apps/kiezradar/models.py

```diff
@@ -100,8 +101,24 @@ class SearchProfile(models.Model):
         blank=True,
     )
 
+    def save(self, update_fields=None, *args, **kwargs):
+        """Custom save() to add the next unused number per user"""
+        if self.number is None:
+            latest = None
+            try:
+                latest = SearchProfile.objects.filter(user=self.user).latest("number")
+            except SearchProfile.DoesNotExist:
+                pass
+            self.number = latest.number + 1 if latest else 1
```

Comment:
here we check if self has no id, which means is new, and give it the generated name. Which also means we don't need to have the name field optional